src/main/java/uk/gov/gchq/palisade/rule/Rules.java

```diff
@@ -20,6 +20,7 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
+import uk.gov.gchq.palisade.Generated;
 import uk.gov.gchq.palisade.ToStringBuilder;
 import uk.gov.gchq.palisade.jsonserialisation.JSONSerialiser;
 
```

Comment:
needs the hashcode and tostring updating. Also the setters, getters and mutators that follow the standard pattern should be updated. The equals method here is more complex than the generated one, so probably should just be updated to use the .equals methods rather than the equalsBuilder

Reply:
Note, did not factor in the Json serialising/deserialising into the equals method that was in the original equals.  I am working with the understanding that by the time we are in the equals methods, we have Rule objects stored in the map.   We should not have to concern ourselves with what the JSON equivalent of the object.  Did I miss something here?   Also note the addRules may have a bug in the way this it is set-up.  If I add a rule for one that is already in the map, I can overwrite it.  If this is not allowed, I should check to see if each of the rules is already in the map and possibly throw an error.